dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/ResultTable.scala

```diff
@@ -30,7 +32,7 @@ class ResultTable(
       res.map { r =>
         val pathToFirstNode = r.path.slice(0, r.path.map(_.node).indexOf(first.node))
         val completePath    = pathToFirstNode ++ (first +: remainder)
-        r.copy(path = completePath)
+        r.copy(path = Vector(completePath.head) ++ completePath.tail)
```

Comment:
Is this to make a deep copy of the path? If so there is also `completePath.map(_.copy())`

Reply:
Good catch. This is more of a refactoring artifact. Should be possible to just say `completePath` here.

Reply:
Ok, no, this needs to be a copy.